[Feature] Create Renewal/Replacement Request Components

EmilioMena opened the pull request:
## Notion ticket link
<!-- Please replace with your ticket's URL -->
[Notion Ticket](https://www.notion.so/uwblueprintexecs/Renewal-Replacement-Request-pages-Pending-status-a7b80f2bad1d493ca5835cd32ae4c77b)


<!-- Give a quick summary of the implementation details, provide design justifications if necessary -->
## Implementation description
Tag team with @chrischan325 

Followed [figma designs](https://www.figma.com/file/GjxN42wF0Qr7tbtaBV4Xoj/RCD-%2F-RCD-Experience) as much as possible
![image](https://user-images.githubusercontent.com/55164714/126101605-cedf80a3-1d0c-4181-9142-c956bc1d954f.png)

Created:
- `PersonalInformationCard`
- `DoctorInformationCard`
- `ReasonForReplacementCard`
- `PaymentInformationCard`

- Base `Card` component that all other Cards use
- No state management

<!-- Catch all section that could be used to draw attention to anything the reviewers should keep in mind, substantial parts of your PR, anything you'd like a second opinion on, things that will be fixed in the future, or things that were left out -->
## Notes
* I'm personally still pretty new to the front-end world so would definitely appreciate any comments/improvements in code structure or design


## Checklist
- [X] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

OustanDing commented on components/internal/Card.tsx:
```diff
@@ -0,0 +1,19 @@
+import { Box, BoxProps } from '@chakra-ui/react';
+
+export default function Card(rest: BoxProps) {
```
Can we call this `props`?

OustanDing replied:
Please add JSDoc comment explaining that this is a custom component with custom styling

OustanDing commented on components/internal/Card.tsx:
```diff
@@ -0,0 +1,19 @@
+      flexDirection="column"
+      alignItems="flex-start"
+      padding="20px 24px 24px"
+      // position="absolute"
```
Remove

OustanDing commented on components/requests/DoctorInformationCard.tsx:
```diff
@@ -0,0 +1,66 @@
+import { Box, HStack, Text, Divider, SimpleGrid, Link } from '@chakra-ui/react';
+import Card from '@components/internal/Card';
+
+type doctorInformationProps = {
```
```suggestion
type DoctorInformationProps = {
```
Please use PascalCase for types!

OustanDing commented on components/requests/DoctorInformationCard.tsx:
```diff
@@ -0,0 +1,66 @@
+  readonly postalCode: string;
+};
+
+export default function DoctorInformationCard(props: doctorInformationProps) {
```
Instead of passing every single physician attribute as its own individual prop, we want to pass in the physician object as a whole as the prop. You can find the physician type in `lib/graphql/types.ts`

OustanDing commented on components/requests/DoctorInformationCard.tsx:
```diff
@@ -0,0 +1,66 @@
+
+export default function DoctorInformationCard(props: doctorInformationProps) {
+  return (
+    <Card w="738px" h="346px">
```
We don't want to use hardcoded values for the width and height since we have t he 12-column grid layout. It's kind of hard to find in the Figma, but if you go to RCD Experience > Hifi Screens > Requests and toggle on Layout Grids using Ctrl+Shift+4, you will see the layout grid on this screen. To take advantage of this, please use the `GridItem` component, which you can see in examples on other pages of our app.

EmilioMena replied:
Good call! Thanks for the suggestions

OustanDing commented on components/requests/DoctorInformationCard.tsx:
```diff
@@ -0,0 +1,66 @@
+            {props.city} {props.province}
+          </Text>
+          <Text textStyle="body-regular">Canada</Text>
+          <Text textStyle="body-regular"> {props.postalCode}</Text>
```
```suggestion
          <Text textStyle="body-regular">{props.postalCode}</Text>
```

OustanDing commented on components/requests/PaymentInformationCard.tsx:
```diff
@@ -0,0 +1,90 @@
+import Card from '@components/internal/Card';
+import { PaymentType } from '@lib/graphql/types';
+
+type paymentInformationProps = {
```
Same here: PascalCase for types + pass in a single object as the prop

OustanDing commented on components/requests/DoctorInformationCard.tsx:
```diff
@@ -0,0 +1,66 @@
+import { Box, HStack, Text, Divider, SimpleGrid, Link } from '@chakra-ui/react';
+import Card from '@components/internal/Card';
```
Comments for imports

OustanDing commented on components/internal/Card.tsx:
```diff
@@ -0,0 +1,19 @@
+import { Box, BoxProps } from '@chakra-ui/react';
```
Comments for imports

OustanDing commented on components/requests/PaymentInformationCard.tsx:
```diff
@@ -0,0 +1,90 @@
+import { Box, HStack, Text, Divider, SimpleGrid, Link, VStack } from '@chakra-ui/react';
+import Card from '@components/internal/Card';
+import { PaymentType } from '@lib/graphql/types';
```
Comments for imports

OustanDing commented on components/requests/PersonalInformationCard.tsx:
```diff
@@ -0,0 +1,85 @@
+import { Box, HStack, VStack, Text, Divider, Badge, Link } from '@chakra-ui/react';
+import Card from '@components/internal/Card';
+
+type personalInformationProps = {
+  readonly name: string;
+  readonly userId: string;
+  readonly mostRecentAPP: string;
+  readonly expiryDate: string;
+  readonly email: string;
+  readonly phoneNumber: string;
+  readonly province: string;
+  readonly city: string;
+  readonly address: string;
+  readonly postalCode: string;
+};
+
+export default function PersonalInformationCard(props: personalInformationProps) {
```
Same as above files

OustanDing commented on components/requests/PersonalInformationCard.tsx:
```diff
@@ -0,0 +1,85 @@
+
+export default function PersonalInformationCard(props: personalInformationProps) {
+  return (
+    <Card w="521px" h="573px">
```
Use `GridItem` for responsive width

OustanDing commented on components/requests/ReasonForReplacementCard.tsx:
```diff
@@ -0,0 +1,53 @@
+import { Box, HStack, Text, Divider, SimpleGrid, Link } from '@chakra-ui/react';
```
Please see files above for general comments

EmilioMena commented on components/requests/PaymentInformationCard.tsx:
```diff
@@ -0,0 +1,90 @@
+import { PaymentType } from '@lib/graphql/types'; // PayementType Enum
+import { MouseEventHandler } from 'react'; // React
+
+type PaymentInformationProps = {
```
@OustanDing couldn't pass in an object as props here since we currently don't store these checkout details anywhere.

angeladietz commented on components/requests/PersonalInformationCard.tsx:
```diff
@@ -0,0 +1,79 @@
+        )}
+        <Box>
+          <Text textStyle="body-regular">{applicant.addressLine1}</Text>
+          <Text textStyle="body-regular">{applicant.city}</Text>
+          <Text textStyle="body-regular">Canada</Text>
+          <Text textStyle="body-regular">{applicant.province}</Text>
```
Missing postal code, and province should be on the same line as the city. Right now it would look like this:
```
Waterloo
Canada
ON
```
but I think we want it to look like this:
```
Waterloo ON
Canada
H2J 3D8
```

OustanDing replied:
Good catch!

angeladietz commented on components/internal/PermitHolderInfoCard.tsx:
```diff
@@ -0,0 +1,50 @@
+    >
+      <Flex w="100%" justifyContent="flex-start" alignItems="center">
+        {typeof header == 'string' ? (
+          <Text textStyle="display-small-semibold">{header}</Text>
```
All of the `Text` elements are missing the `as` prop, please add it to each of them. You can see documentation about it here: https://chakra-ui.com/guides/as-prop. It basically helps browsers generate semantic html which helps screen readers work better. Checkout [verify-identity.tsx](https://github.com/uwblueprint/richmond-centre-for-disability/blob/6189557c1facd578ec1e9278c2fd5017ce8b80cb/pages/verify-identity.tsx) for examples.

angeladietz reviewed (changes requested):
Looks great overall! Just left a few small comments

OustanDing commented on components/requests/PaymentInformationCard.tsx:
```diff
@@ -0,0 +1,90 @@
+            <Text textStyle="body-bold">Shipping Address</Text>
+          </Box>
+          <Box>
+            <Text textStyle="body-regular">{props.shippingAddress}</Text>
```
True. I'll bring this up with @carelynntsai

OustanDing commented on components/requests/PaymentInformationCard.tsx:
```diff
@@ -0,0 +1,90 @@
+            <Text textStyle="body-bold">Billing Address</Text>
+          </Box>
+          <Box>
+            <Text textStyle="body-regular">{props.billingAddress}</Text>
```
Yup, noted!

OustanDing commented on components/requests/DoctorInformationCard.tsx:
```diff
@@ -0,0 +1,61 @@
+        <InfoSection title={`MSP #`}>{String(physician.mspNumber)}</InfoSection>
+        <InfoSection title={`Phone`}>{physician.phone}</InfoSection>
+        <InfoSection title={`Address`}>
+          {physician.addressLine1}
```
@EmilioMena can we just have this be 
```suggestion
          {physician.addressLine1} {physician.addressLine2}
```

OustanDing commented on components/requests/PersonalInformationCard.tsx:
```diff
@@ -0,0 +1,79 @@
+          </Text>
+        )}
+        <Box>
+          <Text textStyle="body-regular">{applicant.addressLine1}</Text>
```
Can be addressed in the same way as `physician.addressLine2` above

EmilioMena commented:
@angeladietz thanks for the review, you had some really good points! 🚀 

I addressed your comments in the latest commit, let me know what you think!

OustanDing reviewed (approved):
LGTM! Great work :D and thanks a lot for the review @angeladietz !

angeladietz reviewed (approved):
Looks good to me as well 🚀
